Fixed snapshot Makefile, so that it builds on osx.

ianchakeres opened the pull request:
Make fails on osx with the following error:

```
$ make container-quick
cp _output/bin/snapshot-controller deploy/docker/controller
cp _output/bin/snapshot-provisioner deploy/docker/provisioner
# Copy the root CA certificates -- cloudproviders need them
cp -r --remove-destination deploy/ca-certificates/* deploy/docker/controller/
cp: illegal option -- -
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
make: *** [container-quick] Error 64
```

This error occurs because osx's `cp` command does not support the `--remove-destination`. 

Additionally, osx `cp` command does not support `-r` and instead it supports `-R` for recursive copy.

ianchakeres committed:
Fixed snapshot Makefile, so that it builds on osx.

ianchakeres commented:
/assign @tsmetana

tsmetana commented:
Yes. The --remove-destination was there to ensure the files get overwritten... I didn't realize none of the options are POSIX...

/lgtm

ianchakeres committed:
Removed dangling symlinks in ca-certificates to locations outside this repo, and added -f option to cp command to force overwrite

ianchakeres commented:
@tsmetana in the latest commit, I added `-f` option to `cp` to overwrite any existing files.

I also removed several symlinks in ca-certificates, that were pointing to files not included in this repo. These symlinks were resulting in `cp` failures.

```
$ cp -Rf deploy/ca-certificates/* deploy/docker/controller/.
cp: deploy/ca-certificates/etc/ssl/certs: No such file or directory
cp: deploy/ca-certificates/etc/pki/java/cacerts: No such file or directory
cp: deploy/ca-certificates/etc/pki/ca-trust/source/ca-bundle.legacy.crt: No such file or directory
[ichakeres@ichakeres-ltm4 snapshot (osx-fix-snapshot-make)]$ ls -alh deploy/ca-certificates/etc/ssl/certs
lrwxr-xr-x  1 ichakeres  wheel    16B Mar 10 21:13 deploy/ca-certificates/etc/ssl/certs -> ../pki/tls/certs
[ichakeres@ichakeres-ltm4 snapshot (osx-fix-snapshot-make)]$ ls -alh deploy/ca-certificates/etc/pki/java/cacerts
lrwxr-xr-x  1 ichakeres  wheel    40B Mar 10 21:13 deploy/ca-certificates/etc/pki/java/cacerts -> /etc/pki/ca-trust/extracted/java/cacerts
[ichakeres@ichakeres-ltm4 snapshot (osx-fix-snapshot-make)]$ ls -alh deploy/ca-certificates/etc/pki/ca-trust/source/ca-bundle.legacy.crt
lrwxr-xr-x  1 ichakeres  wheel    59B Mar 10 21:13 deploy/ca-certificates/etc/pki/ca-trust/source/ca-bundle.legacy.crt -> /usr/share/pki/ca-trust-legacy/ca-bundle.legacy.default.crt
```

rootfs commented:
/assign @tsmetana

tsmetana commented:
/lgtm
